MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -149,12 +149,26 @@ class MSALNativeAuthBaseController {
         completion(response, error)
     }
 
-    func performRequest<T>(_ request: MSIDHttpRequest, context: MSIDRequestContext) async -> Result<T, Error> {
+    func performRequest<T: MSALNativeAuthResponseCorrelatable>(
+        _ request: MSIDHttpRequest,
+        context: MSALNativeAuthRequestContext
+    ) async -> Result<T, Error> {
         return await withCheckedContinuation { continuation in
-            request.send { result, error in
+            request.send { [weak self] result, error in
                 if let error = error {
+                    if let correlationId = self?.extractCorrelationIdFromUserInfo((error as NSError).userInfo) {
+                        context.setServerCorrelationId(UUID(uuidString: correlationId))
```

Comment:
can you add a comment for each code block explaining when it should be called?

Reply:
Done